lib/orderbook/MatchingEngine.ts

```diff
@@ -152,16 +157,20 @@ class MatchingEngine {
     return matchingResult;
   }
 
-  public removeOwnOrder = (orderId: string): orders.StampedOwnOrder | null => {
-    return this.priorityQueues.buyOrders.removeOne(order => order.id === orderId) ||
-      this.priorityQueues.sellOrders.removeOne(order => order.id === orderId);
+  public removeOwnOrder = (orderId: string): StampedOwnOrder | undefined => {
+    return (this.priorityQueues.buyOrders.removeOne(order => order.id === orderId) ||
+      this.priorityQueues.sellOrders.removeOne(order => order.id === orderId)) as StampedOwnOrder | undefined;
   }
 
-  public removePeerOrders = (predicate: Function): orders.StampedPeerOrder[] => {
+  public removePeerOrders = (hostId: number): StampedPeerOrder[] => {
```

Comment:
If later we'll need to remove multiple peer orders by a different param, we'll need another function for that. I used a predicate function instead of a regular param to make the interface more open.

Reply:
I see, I guess I'm thinking it's better to have the predicate logic in the method itself. It certainly seems better for testing, that's how the previous bug slipped through tests. We can create more methods down the road. Another option might be to define the predicates in one place, so that the tests use exactly the same function as the main code.

Reply:
The bugs was slipped because we didn't do yet integration for that section from `OrderBook`, only to the underlying `MatchingEngine`.
Defining the predicates in a separate place won't be a good idea IMO.

We can keep it like this if you prefer, and reassess the design later.